app/tests/spec/views/sign_in_unblock.js

```diff
@@ -142,7 +142,7 @@ define(function (require, exports, module) {
       describe('with an empty code', () => {
         beforeEach(() => {
           view.$('#unblock_code').val('');
-          return view.validateAndSubmit();
+          return view.validateAndSubmit().then(assert.fail, () => {});
```

Comment:
This seems like it fundamentally changes the behaviour of the `beforeEach`, making failure the successful path. What is the rationale for that and is there anything different we can do to make it unnecessary? At the very least it warrants an explanatory comment I reckon.

Reply:
Wonder if it would be more apparent what is going on if we removed the `beforeEach` block here.